spec/formatting.md

```diff
@@ -197,16 +197,27 @@ the following steps are taken:
 
 1. If the _expression_ includes an _operand_, resolve its value.
    If this fails, use a _fallback value_ for the _expression_.
-2. Based on the _function_ starting sigil and _name_,
-   find the appropriate function implementation from the _function registry_.
-   If the registry does not define an implementation for this _name_,
-   emit an Unknown Function error
+2. Resolve the _identifier_ of the _function_ and, based on the starting sigil,
+   find the appropriate function implementation to call.
+   If the implementation cannot find the function, emit an Unknown Function error
    and use a _fallback value_ for the _expression_.
-3. Resolve the _option_ values to a mapping of string identifiers to values.
+
+   An implementation MAY emit an Unknown Function error if the implementation    
+   does not support the _namespace_ of the _identifier_
+   or the _namespace_ does not support the named _function_.
+
+   Implementations are not required to implement _namespaces_ or installable
+   _function registries_.
+
+4. If the _expression_ includes _options_, resolve the _options_ to a mapping
+   of string identifiers to values.
    For each _option_:
-   - If its right-hand side successfully resolves to a value,
-     bind the _name_ of the _option_ to the resolved value in the mapping.
-   - Otherwise, do not bind the _name_ of the _option_ to any value in the mapping.
+   - Resolve the _identifier_ of the _option_.
+   - If the _option_'s _identifier_ already exists in the resolved mapping of _options_,
+     emit a Duplicate Option Name error.
+   - If the _option_'s right-hand side successfully resolves to a value,
+     bind the _identifier_ of the _option_ to the resolved value in the mapping.
+   - Otherwise, do not bind the _identifier_ of the _option_ to any value in the mapping.
 4. Call the function implementation with the following arguments:
```

Comment:
Good callout

Reply:
The appropriate-error emission is already taken care of during the resolution of the right-hand value, in Variable Resolution. So if it's noted here, we should not say something like "emit an error" here because that would mean that it needs to be emitted twice.

Reply:
@aphillips Our messages crossed paths; see my comment above for a problem this language introduces.

Reply:
See change below.